### PR TITLE
memtx: fix potential NULL dereference in memtx_space_ephemeral_delete

### PR DESCRIPTION
**Pull request by drewdzzz**
Now, delete in ephemeral space is obviously incorrect - if we try to delete a tuple, which is not present in index, NULL dereference will happen. Fortunately, ephemeral spaces are used for internal purposes only, so, most likely, this never happens. Let's fix this part not to confuse code analyzers.

Closes https://github.com/tarantool/security/issues/38

### CODE REVIEW

**Comment by CuriousGeorgiy** [comment]
IMHO, this patch is wrong, we should instead add an assertion and triage this marker as FP (already wrote about this in the issue).

File: src/box/memtx_space.c [hunk 1]
```diff
@@ -690,11 +690,12 @@ memtx_space_ephemeral_delete(struct space *space, const char *key)
...
 		return -1;
-	if (old_tuple != NULL &&
-	    memtx_space->replace(space, old_tuple, NULL,
-				 DUP_REPLACE, &old_tuple) != 0)
-		return -1;
-	tuple_unref(old_tuple);
+	if (old_tuple != NULL) {
+		if (memtx_space->replace(space, old_tuple, NULL,
+					 DUP_REPLACE, &old_tuple) != 0)
+			return -1;
```
[the remark this answers is not in the document]
Comment (locker) [reply src/box/memtx_space.c:696]: If `replace` returns 0, it doesn't set `old_tuple` so mem leak is impossible.

**Review by locker — APPROVED** [review]
IMO returning 0 and doing nothing if the key isn't found looks like a valid behavior, i.e. I find this patch correct. Whether the caller (i.e. SQL) actually expects this or they had rather this function fail in this case is a different question. Okay, let's ask @ImeevMA.

TBH I think that we shouldn't have introduced "ephemeral" spaces in the first place (we could implement them completely in SQL without adding new `struct space` methods, using the BPS tree container only) so I don't really care how exactly we suppress this SVACE warning.

**Comment by ImeevMA** [comment]
I already gave my answer: https://github.com/tarantool/security/issues/38#issuecomment-1425465846

TL DR: I think the issue should be fixed for the master, but I don't see the need to fix it in version 2.10. Although there are no problems with fixing there. If @locker thinks the patch is correct, I see no reason to protest.

**Review by ImeevMA — APPROVED** [review]
I agree with @locker.

**Comment by locker** [comment]
Cherry-picked to 2.10.

**Comment by CuriousGeorgiy** [comment]
@ImeevMA my point was that a situation where `old_tuple` is `NULL` is impossible.

**Comment by ImeevMA** [comment]
This is true right now, however there is an issue https://github.com/tarantool/tarantool/issues/3375 which suggests making it possible to access ephemeral methods from Lua. I think this issue will be rejected at the end, but who knows.
